content/AbstractClassesAndMethods/AbstractClassesAndMethods.q.md

```diff
@@ -0,0 +1,2106 @@
+
+ - It removes the `Person` class definition.
+
+ - It hides the `Person` class implementation.
+
+ - It creates an instance of the `Person` class.
+
+ - It defines the abstract class named `Person`.
+
+ - It sets the value of the `Person` class to abstract.
+
+
+/// type=MS, answer=[3,4]
+
+In the definition `abstract public function display();` on line 40 of `Person.php`, what does the `abstract` keyword do?
+
+ - It removes the `display()` method definition.
+
+ - It hides the `display()` method implementation.
+
+ - It defines the abstract method named `display()`.
+
+ - It declares the `display()` method without a body.
+
+ - It sets the value of the `display()` method to abstract.
+
+
+/// type=MS, answer=[2,4]
+
+Which statements correctly describe `class Student extends Person { }` on line 2 of `Student.php`?
+
+ - It is a definition of the `Student` class that replaces the `Person` class.
+
+ - It is a definition of the `Student` class as the child class of the `Person` class.
+
+ - It is a definition of the `Person` class as the child class of the `Student` class.
+
+ - It creates the `Student` class that inherits all the `public` and `protected` properties and methods of the `Person` class. 
+
+ - It creates the `Person` class that inherits all the `public` and `protected` properties and methods of the `Student` class. 
+
+
+/// type=SS, answer=[2]
+
+Which statement best describes `parent::__construct($name, $age);` on line 8 of `Student.php`?
+
+ - It creates the `__construct()` method of the parent class `Person` with the parameters `$name` and `$age` inside the child class `Student`.
+
+ - It calls the `__construct()` method of the parent class `Person` passing the arguments `$name` and `$age` inside the child class `Student`.
+
+ - It returns the arguments `$name` and `$age` from the `__construct()` method of the parent class `Person` inside the child class `Student`.
+
+ - It defines the `__construct()` method of the parent class `Person` with the parameters `$name` and `$age` inside the child class `Student`.
+
+ - It removes the arguments `$name` and `$age` from the `__construct()` method of the parent class `Person` inside the child class `Student`.
+
+
+/// type=SS, answer=[1]
+
+In the `echo` statement on line 19 of `Student.php`, what does `parent::getName()` do?
+
+ - It calls the `getName()` method of the parent class `Person` inside the child class `Student`.
+
+ - It removes the `getName()` method of the parent class `Person` from the child class `Student`.
+
+ - It returns the `getName()` method of the parent class `Person` inside the child class `Student`.
+
+ - It creates the `getName()` method of the parent class `Person` inside the child class `Student`.
+
+ - It defines the `getName()` method of the parent class `Person` inside the child class `Student`.
+
+
+/// type=MS, answer=[3,4]
+
+Which statements correctly describe the `display()` method of the `Student` class?
```

Comment:
What if we specify the line number of the method for this question?

Reply:
I think this no longer necessary since there is only one display method in the `Student` class. Please check. Thanks.